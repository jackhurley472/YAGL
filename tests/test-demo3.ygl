/* Every single variable is named foo */

Graph foo;
{
   Node bar("Michael1");
   foo + bar;
   Node bar2("Michael2");
   foo + bar2;
   foo: bar ->1 bar2;
   int i = 0;
   while (i < 3) {
	   if (i == 2) { 
		   Graph foo;
		   Node bar("Finished");
		   foo + bar;
		   foo: bar2 -> bar;
		   foo + bar2;
		   foo: bar2 ->5 bar;
		   printGraph(foo);
	   } else {
		   if (i == 1) {
			   Graph foo;
			   Node bar("Michael5");
			   foo + bar;
			   Node bar2("Michael6");
		   	   foo + bar2;
		   	   foo: bar -> bar2;
		   	   printGraph(foo);
		   } else {
			   Graph foo;
			   Node bar("Michael3");
			   foo + bar;
			   Node bar2("Michael4");
		   	   foo + bar2;
		   	   foo: bar -> bar2;
		   	   printGraph(foo);
		   }
	   }
	   i = i + 1;
   }
}
printGraph(foo);
