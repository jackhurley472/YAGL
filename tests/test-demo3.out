============== Graph Print ===============
	Stats: 	2	2	1

Nodes:
Node (2) : Michael3 --- Node (3) : Michael4

Edges:
[(2) : Michael3] ---(1)---> [(3) : Michael4]
============ End Graph Print =============
============== Graph Print ===============
	Stats: 	2	2	1

Nodes:
Node (4) : Michael5 --- Node (5) : Michael6

Edges:
[(4) : Michael5] ---(1)---> [(5) : Michael6]
============ End Graph Print =============
Nodes don't exist in given graph.
============== Graph Print ===============
	Stats: 	2	2	1

Nodes:
Node (6) : Finished --- Node (1) : Michael2

Edges:
[(1) : Michael2] ---(5)---> [(6) : Finished]
============ End Graph Print =============
============== Graph Print ===============
	Stats: 	2	2	1

Nodes:
Node (0) : Michael1 --- Node (1) : Michael2

Edges:
[(0) : Michael1] ---(1)---> [(1) : Michael2]
============ End Graph Print =============
